### PR TITLE
feat(webdriver): add support for NO_PROXY environment variable

### PR DESCRIPTION
**Pull request by dsebastien**
See discussion in #3046

### CODE REVIEW

File: bin/webdriver-manager [hunk 1]
```diff
@@ -173,15 +173,47 @@ if (!fs.existsSync(argv.out_dir) || !fs.statSync(argv.out_dir).isDirectory()) {
...
-    return process.env.HTTPS_PROXY || process.env.https_proxy || process.env.HTTP_PROXY || process.env.http_proxy;
-  } else if (protocol === 'http:') {
-    return process.env.HTTP_PROXY || process.env.http_proxy;
-  }
+var resolveProxy = function (fileUrl) {
+    var parsedUrl = url.parse(fileUrl);
+    var protocol = parsedUrl.protocol;
+    var hostname = parsedUrl.hostname;
+
+    // NO_PROXY environment variable
```
Comment (cnishina) [inline bin/webdriver-manager]: See comment below to clean this up.

File: bin/webdriver-manager [hunk 2]
```diff
@@ -173,15 +173,47 @@ if (!fs.existsSync(argv.out_dir) || !fs.statSync(argv.out_dir).isDirectory()) {
...
+
+    // will be set to true if the hostname of the fileUrl ends with any of the noProxyTokens
+    var noProxyMatchFound = false;
+
+    if (argv.proxy) {
+        // if a proxy was explicitly specified, then it must be used
+        return argv.proxy;
+    } else {
+        // first we check if the fileUrl hostname part does not end with one of the NO_PROXY environment variable's hostnames/domain names
+        noProxyTokens.forEach(function (noProxyToken) {
```
Comment (cnishina) [inline bin/webdriver-manager]: Let's clean up and condense the noProxy stuff here...

```
var noProxy = process.env.NO_PROXY || process.env.no_proxy;
if (noProxy) {
  // remove spaces, split,
  // forEach
    // check if matches and if it does, return null
}
```

**Comment by dsebastien** [comment]
Thanks for the review; I've updated the PR.

**Comment by cnishina** [comment]
closed with #de4b33eaab1546d0ef3a746cfd222e80f0ec28a1

**Pull request closed without merging**